src/vs/workbench/contrib/tasks/electron-browser/task.contribution.ts

```diff
@@ -2195,16 +2198,27 @@ class TaskService extends Disposable implements ITaskService {
 		if (!this.canRunCommand()) {
 			return;
 		}
+		if (arg === 'terminateAll') {
+			this.terminateAll();
+			return;
+		}
 		let runQuickPick = (promise?: Promise<Task[]>) => {
 			this.showQuickPick(promise || this.getActiveTasks(),
 				nls.localize('TaskService.tastToTerminate', 'Select task to terminate'),
 				{
 					label: nls.localize('TaskService.noTaskRunning', 'No task is currently running'),
-					task: null
+					task: undefined
 				},
-				false, true
+				false, true,
+				{
+					label: nls.localize('TaskService.terminateAllRunningTasks', 'All running tasks'),
+					task: null
+				}
 			).then(task => {
 				if (task === undefined || task === null) {
+					if (task === null) {
```

Comment:
Instead of having this null or undefined check, I'd prefer that `showQuickPick` be refactored to return the quick pick entry instead of the task. That way you can set an ID on the quick pick entry and check for it here.